endpoint.go

```diff
@@ -303,14 +306,16 @@ func (ep *endpoint) Join(containerID string, options ...EndpointOption) (*Contai
 		}
 	}
 
-	err = sb.SetGateway(joinInfo.gw)
-	if err != nil {
-		return nil, err
-	}
+	if primary {
```

Comment:
once a primary is set, can it be changed to another endpoint ?
The way the current change behaves, we can have multiple endpoints owning the primary flags.
So, either we should not allow multiple endpoints with primary flag (or) provide a way to `unset`
primary in the first endpoint holding the primary key and create a new one to be primary.